### DISCUSSION TITLE
RLS Policy based on values from the same table where the policy is applied.

### DISCUSSION THREAD

**Discussion by 5thComrade** [supabase/supabase#16079] [Questions] [answered]
Hi Guys,
Disclaimer: I'm kinda new to RLS.

I will try to explain my situation with an example.
Let's say I have a Users table. I have enabled RLS on this table. The Users table holds the details of the users of my application, some of the columns on this table include, name, email, phone and role. Role can either be 'customer' or 'admin'.

I want to define a RLS policy on SELECT.
If the logged in user is a 'customer', then this user should only have access to their data. However if the logged in user is an 'admin', they should have access to any user details including theirs. I am struggling to define an RLS policy for this case.

I tried something like this 

user_id = auth.uid() or exists (select 1 from users u where u.user_id = auth.uid() and u.role = 'admin')

but this did not work. It would be really great if someone could help me out here.

**Comment by GaryAustin1** [comment, marked as answer]
You need to use a security definer function to do the select on a table if you are in the select policy for that table.  That function bypasses RLS.
https://supabase.com/docs/guides/auth/row-level-security#policies-with-security-definer-functions
